<?xml version="1.0"?>
<launch>

	# Input arguments
	<arg name="viz_DS_path"   default="true"/>
 	<arg name="sim"           default="false"/>
 	<arg name="letter"        default="Angle"/>

	# Options for letters:
	# - Angle
	# - C
	# - Khamesh 	

   # Load SEDS motion generator for "Approaching" in writing task
	<group ns="ds1">
	    <param name="input_topic_name"                  value="/lwr/ee_pose"/>
	    <param name="input_target_topic_name"           value="/ds1/desired_target"/>
	    <param name="dynamic_target"                    value="true"/>
	    <param name="output_topic_name"                 value=""/>
	    <param name="output_filtered_topic_name"        value="/ds1/desired_velocity_filtered"/>
	    <param name="publish_DS_path"                   value="false"/>	    
	    <rosparam command="load" file="$(find ds_motion_generator)/config/learned_DS/seDS/Curve_go_right.yml"/>
	    <node name="ds_node" pkg="ds_motion_generator"  type="seDS_motionGenerator_node"/>
	</group>

	<group if="$(arg viz_DS_path)">
	  <group ns="ds1_path">
	      <param name="input_topic_name"                  value="/lwr/ee_pose"/>
	      <param name="input_target_topic_name"           value="/ds1/desired_target"/>
	      <param name="dynamic_target"                    value="true"/>
	      <param name="output_topic_name"                 value=""/>
	      <param name="output_filtered_topic_name"        value=""/>    
	      <param name="publish_DS_path"                   value="true"/>
	      <rosparam command="load" file="$(find ds_motion_generator)/config/learned_DS/seDS/Curve_go_right.yml"/>
	      <node name="ds_node" pkg="ds_motion_generator"  type="seDS_motionGenerator_node" />
	  </group>
	</group>


	# Load LPV-DS motion generator for "Writing" in inspection task
	<group ns="ds2">
	    <param name="input_topic_name"                  value="/lwr/ee_pose"/>
	    <param name="input_target_topic_name"           value="/ds2/desired_target"/>
	    <param name="dynamic_target"                    value="true"/>
	    <param name="output_topic_name"                 value=""/>
	    <param name="output_filtered_topic_name"        value="/ds2/desired_velocity_filtered"/>
	    <param name="publish_DS_path"                   value="false"/>
	    <rosparam command="load" file="$(find ds_motion_generator)/config/learned_DS/lpvDS/KUKA-$(arg letter)Shape.yml"/>
	    <node name="lpvDS_node" pkg="ds_motion_generator"  type="lpvDS_motionGenerator_node" />
	</group>

	<group if="$(arg viz_DS_path)">
	  <group ns="ds2_path">
	      <param name="input_topic_name"                  value="/lwr/ee_pose"/>
	      <param name="input_target_topic_name"           value="/ds2/desired_target"/>
	      <param name="dynamic_target"                    value="true"/>
	      <param name="output_topic_name"                 value=""/>
	      <param name="output_filtered_topic_name"        value=""/>    
	      <param name="publish_DS_path"                   value="true"/>
	      <rosparam command="load" file="$(find ds_motion_generator)/config/learned_DS/lpvDS/KUKA-$(arg letter)Shape.yml"/>
	      <node name="lpvDS_node" pkg="ds_motion_generator"  type="lpvDS_motionGenerator_node" />
	  </group>
	</group>


	# Load Motion Planning node for Table Task
	<group>
		  <param name="input_pose_topic_name"              value="/lwr/ee_pose"/>
		  <param name="input_ds1_topic_name"               value="/ds1/desired_velocity_filtered"/>
		  <param name="input_ds2_topic_name"               value="/ds2/desired_velocity_filtered"/>
	      <param name="output_vel_topic_name"              value="/lwr/joint_controllers/passive_ds_command_vel"/>
	      <param name="output_pick_topic_name"             value="/ds1/desired_target"/>
	      <param name="output_write_topic_name"             value="/ds2/desired_target"/>
	      <param name="sim"                                value="$(arg sim)"/>
	      <rosparam command="load" file="$(find kuka_lpvds_tasks)/config/writingTask-$(arg letter)Shape.yml"/>
		  <node name="writingTask_motionPlanner_node" pkg="kuka_lpvds_tasks"  type="writingTask_motionPlanner_node" output="screen"/>
	</group>
	
	# This node published the desired orientation at the target
	<node pkg="rostopic" type="rostopic" name="angle_setting" args="pub /lwr/joint_controllers/passive_ds_command_orient geometry_msgs/Quaternion '{x: -0.0426, y: 0.998, z:  0.0039, w: -0.0356}' -r 100" />
</launch>
